paasta_tools/cli/cmds/get_latest_deployment.py

```diff
@@ -53,13 +51,7 @@ def paasta_get_latest_deployment(args):
     soa_dir = args.soa_dir
     validate_service_name(service, soa_dir)
 
-    git_url = get_git_url(
-        service=service,
-        soa_dir=soa_dir,
-    )
-    remote_refs = list_remote_refs(git_url)
-
-    _, git_sha = get_latest_deployment_tag(remote_refs, deploy_group)
+    git_sha = get_currently_deployed_sha(service=service, deploy_group=deploy_group, soa_dir=soa_dir)
```

Comment:
This raises a `NoDeploymentsAvailableException` if the service isn't deployed yet. You should catch that instead of checking `not git_sha` later